lib/processor.js

```diff
@@ -145,6 +156,24 @@ function preprocess(text) {
  * @returns {Message[]} A flattened array of messages with mapped locations.
  */
 function postprocess(messages) {
+
+    /**
+     * Returns messages which are not in the list of expected errors
+     * @param  {Message[]}  msgs A flattened array of messages with mapped locations.
+     * @returns {Message[]}      An array of messages without expected messages.
+     */
+    function getUnexpectedMessages(msgs) {
+        return msgs.filter(function (msg) {
+            var msgIsExpected = false;
+            expectedErrors.forEach(function (expErr) {
```

Comment:
By "some kind of map" do you mean `Array.prototype.map`?  A nudge in the right direction would help me a lot here.  I had a feeling there's a better way, but I don't have the CS background to know what technique to reach for.